### DISCUSSION TITLE
Annotated JPEG in BI Alert?

### DISCUSSION THREAD

**Discussion by blulite** [VorlonCD/bi-aidetection#349] [Q&A] [answered]
The default pastes the non-annotated image as the Alert thumbnail in BI.   I want to show the annotated version, but we don't seem to have a VARIABLE listed that has the filename for the annotated version (I can see them in my Alerts folder).   Any thoughts, or do I need to add to the code?

Jeff

**Comment by VorlonCD** [comment]
Hi Jeff.  The default Trigger URL > jpeg=[ImagePathEscaped] has the annotations "burned" into the image as long as Aitool > Camera actions > "Merge annotations into images" is checked.

If thats not what you mean or it doesnt help, give a little more detail.

**Reply by blulite** [reply]
Hmmm, not for me.    I've got the "triggers" from BI sending the JPEG to a folder called W:\aiinputs, and my Alert folder is W:\alerts.   [ImagePathEscaped] points to the W:\aiinputs directory and the files located there are the triggered jpegs but without annotations.   The file has an annotated version in W:\alerts, but it has a different filename.

BTW, I use Telegram, and the jpegs come through with the annotations.

**Reply by VorlonCD** [reply]
Ok, I guess I'm not getting the annotations in BI either!   Its a tiny little thumbnail, I never really noticed.

Try using the "Copy alert images to folder:" path and filename.    Just replace any backslashes with %5C, and use [Summary] rather than [SummaryNonEscaped]  Example:

`&jpeg=W:%5calerts%5c[ImageFilenameNoExt]_[Summary].jpg`

**Reply by blulite** [reply]
Typo with [Summary] in filename?

I'm trying to debug, and I notice I'm getting triggers from Motion_A and external in the Alert. Hmmm, I only have the Trigger External checked. I have some other cameras that come back with External, and the overlay shows up. Do you have any ideas? I am banging my head, trying to see where the Motion_A trigger is coming from.

**Comment by VorlonCD** [comment]
No typo for [summary] as far as I can see?   You need to use [summary] vs [summarynonescaped] so it replaces spaces and special chars with URL encoding.

The ones from AITOOL will always be considered "External".

Are you saying that you DO have some showing in BI that have the annotation burned in to the image?    I dont have any.

And it may not be possible??
https://blueirissoftware.com/BlueIris.PDF

"Use of memo, jpeg, flagclip are optional. If supplied, the jpeg will replace any existing hi-res
image for this alert in the alerts folder; **_however it will not replace the thumbnail used in the
database alerts list._** "

I'll see if I can find more time to look into this later.

**Reply by blulite** [reply]
Three items:

1. As for [Summary], do you have it as part of the filename?   I see it in the memo, but does that info become part of a filename?
2. See attached thumbnail for one of my cameras.  These are back to back alerts, without any mods from me between them.  See that one is trigger by "External" only, and it includes the overlay.   The other one includes "External and Motion_A" and it does not include the overlay.   I have no idea where the trigger for Motion_A is coming from, as I have that box unchecked in BI and only the "External" box is checked for Trigger Sources in the Alerts tab.
3. A little feature I discovered is if you reformate the memo field, and put a ':' after the [Detection] format "[Label]: [[Detail]] [confidence]" then you get the vehicle or person icon and you can select a standard view under alerts to just show you vehicles or persons.


<img width="273" alt="Screenshot 2024-06-12 at 4 14 40 PM" src="https://github.com/VorlonCD/bi-aidetection/assets/6198334/ec7c899f-6b5f-4e0d-880c-69a3343ab8ab">

**Comment by VorlonCD** [comment]
nice catch on the icon!

I used [summary] because it was part of the filename for the COPY ALERT IMAGES.     But even that is still not working for me:
![image](https://github.com/VorlonCD/bi-aidetection/assets/7512113/fb175c03-0976-479b-8b9c-8491dd821fa6)

Someone else was having the same issue a few years ago and I believe the solution was to use COPY ALERT IMAGES to directly overwrite the original JPG that BI creates:
https://github.com/VorlonCD/bi-aidetection/issues/260

**Comment by blulite** [comment, marked as answer]
I've got it working.   Why do I care?   If you use the iOS version of the BlueIris app, and select the confirmed images screen with Large icons, you can see the triggered image with the annotation for the trigger.  Plus I can now see the "Vehicle" and "Person" events with icons.   My fix:

1 - Put a ':' after the [Detection] format "[Label]: [[Detail]] [confidence]." Then you get the vehicle or person icon, and you can select a standard view under alerts to just show you vehicles or persons.

2- Created a duplicate camera.   The original camera entry does not trigger, only records video (no save JPEG to aiinput folder, and for alerts, only External alerts

3—The duplicate camera only triggers. I named the duplicate CAM-a (add a "-a", and all it does is trigger on motion and write a JPEG to the aiinput folder. The JPEG will have a "-a" appended to the camera name. Turn off all alerts.

4 - On aiinput.exe, change the input filename to include the "-a" on the jpeg.

So you now have two cameras. The original one functions as normal, and aiinput will alert you on this camera with a JPEG that includes annotation.   The second camera triggers motion and writes the JPEG to aiinput.  You can hide the second camera once you have set it up.

Jeff
